tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -136,6 +149,166 @@ ZTEST_F(ascs_test_suite, test_sink_ase_read_state_idle)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT + 1,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	err = bt_bap_unicast_server_unregister();
+	zassert_equal(err, 0, "unexpected err response %d", err);
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_too_many_sources)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT + 1
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	err = bt_bap_unicast_server_unregister();
+	zassert_equal(err, 0, "unexpected err response %d", err);
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_zero_ases)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		0,
+		0
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	err = bt_bap_unicast_server_unregister();
+	zassert_equal(err, 0, "unexpected err response %d", err);
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_fewer_than_max_ases)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT > 0 ? CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT - 1 : 0,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT > 0 ? CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT - 1 : 0
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	err = bt_bap_unicast_server_unregister();
+	zassert_equal(err, 0, "unexpected err response %d", err);
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, 0, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_unregister_without_register)
+{
+	int err;
+
+	/* Unregister ASCS, as its registered through setup */
+	err = bt_bap_unicast_server_unregister();
+	zassert_equal(err, 0, "unexpected err response %d", err);
+
+	err = bt_bap_unicast_server_unregister();
+
+	zassert_equal(err, -EALREADY, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_unregister_with_ases_in_config_state)
+{
+	const struct test_ase_chrc_value_hdr *hdr;
+	const struct bt_gatt_attr *ase;
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	struct bt_gatt_notify_params *notify_params;
+	uint8_t ase_id;
+	int err;
+
+	if (IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK)) {
+		ase = fixture->ase_snk.attr;
+		ase_id = fixture->ase_snk.id;
+	} else {
+		ase = fixture->ase_src.attr;
+		ase_id = fixture->ase_src.id;
+	}
+
+	zexpect_not_null(ase);
+	zexpect_true(ase_id != 0x00);
+
+	bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
```

Comment:
Missing error check

Reply:
Actually, it is like this because the original tests made by Mariusz are lacking the error check, everywhere (14 instances). I assume this is the case because this is considered a part of the setup of the function, and not a step which is actually being tested. If the test fails because of error in this function call, then there is something gravely faulty, unrelated to the test per se.

You could argue that not checking this here makes the test easier to read, and perhaps also to debug.

Reply:
However, I already added this on another reg function everywhere, so Ill add this too for consistency.

Reply:
> You could argue that not checking this here makes the test easier to read, and perhaps also to debug.

If this fails silently, then debugging is going to be much harder when the test fails :D